### PR TITLE
New package: sourmash (and deps)

### PR DESCRIPTION
**Pull request by luizirber**
### Description

Add sourmash (Python + Rust extension, packaged with maturin) and missing pure-wheel deps (screed, bitstring, cachetools, deprecation).

After a couple of fixes in https://github.com/sourmash-bio/sourmash/pull/2433 I managed to build it out-of-tree, would like to have it available here to make it easier to distribute (while PyPI doesn't support emscripten wheels).

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: docs/project/changelog.md
Comment (ryanking13) [inline docs/project/changelog.md:168]: I think it is okay to add only `sourmash` here, since other packages are pure python packages.

**Review by ryanking13 — COMMENTED** [review]
Cool, thanks @luizirber!

File: docs/project/changelog.md
Comment (luizirber) [reply docs/project/changelog.md]: Got it! I ended up adding them because when I tried to test locally they didn't seem to be read as dependencies unless listed in the `meta.yaml` file, do you have pointers on how to declare them?

**Comment by hoodmane** [comment]
@ryanking13 @rth anyone understand the error here? It looks unrelated to this PR?

**Comment by ryanking13** [comment]
It seemsl like you need to add:

```yaml
requirements:
  run:
    - packaging
```

to meta.yaml file of `deprecation`.

File: docs/project/changelog.md
Comment (ryanking13) [reply docs/project/changelog.md]: You are doing it the right way. We do dependency resolution before downloading a package for in-tree packages, so it has to be explicitly declared in meta.yaml and all dependencies have to be added in-tree (which I want to change in the future https://github.com/pyodide/pyodide/issues/2580.)

**Review by ryanking13 — APPROVED** [review]
Thanks @luizirber! I have some minor comments otherwise LGTM.

File: packages/sourmash/meta.yaml
Comment (luizirber) [reply packages/sourmash/meta.yaml:20]: Yup, will do! I left this PR as draft because the [pyodide-fixing PR](https://github.com/sourmash-bio/sourmash/pull/2433) and CI checks were added after `4.7.0`, but I will release `4.7.1` and fix this URL/sha256

Comment (luizirber) [reply packages/sourmash/meta.yaml]: [sourmash 4.8.0](https://pypi.org/project/sourmash/4.8.0/) just released, I updated the PR to point to PyPI sdist

**Review by hoodmane — APPROVED** [review]
This looks good to me. Any reason you have it marked as a draft?

**Comment by luizirber** [comment]
> This looks good to me. Any reason you have it marked as a draft?

I was waiting for the 4.8.0 release, removed the draft status now =]

**Comment by hoodmane** [comment]
Okay I will merge if CI looks good.

**Comment by luizirber** [comment]
@hoodmane @ryanking13 I think this is ready for merge =)

**Comment by luizirber** [comment]
The merge build failed due to an unrelated error ([building mpfr](https://app.circleci.com/pipelines/github/pyodide/pyodide/5659/workflows/403d9a84-f5c0-4814-bb87-e6d13913a357/jobs/71025?invite=true#step-104-198)?)

**Comment by hoodmane** [comment]
Yeah apparently mpfr's ssl certificate expired or something.

**Pull request merged**
